docs/guide/sap/sap-s4hana.md

```diff
@@ -186,6 +186,75 @@ SAP application servers communicate constantly with the database servers. For pe
+
+```plaintext
+net.ipv4.tcp_mem = 4096 87380 67108864
+net.ipv4.udp_mem = 4096 87380 33554432
+net.ipv4.tcp_rmem = 4096 87380 67108864
+net.ipv4.tcp_wmem = 4096 65536 67108864
+net.core.rmem_default = 33554432
+net.core.wmem_default = 33554432
+net.ipv4.udp_wmem_min = 16384
+net.ipv4.udp_rmem_min = 16384
+net.core.wmem_max = 134217728
+net.core.rmem_max = 134217728
+net.core.busy_poll = 50
+net.core.busy_read = 50
+````
+
+2. Congestion-Based Congestion control for kernels 4.19 and above: Enabling Bottleneck Bandwidth and Round-trip propagation time (BBR) congestion control can often result in better throughput. Add this configuration to ```/etc/sysctl.d/99-azure-congestion-control.conf```:
+
+```plaintext
+net.ipv4.tcp_congestion_control = bbr
+# The standard recommendation when using BBR is to enable the FQ codel, you need to test to see if either fq or pfifo_fast will work better in your deployment.
+net.core.default_qdisc = fq
+```
+
+3. Extra TCP parameters that will usually help with better consistency, throughput: Add these configurations to ```/etc/sysctl.d/99-azure-network-extras.conf``` :
+
+```plaintext
+# For deployments where the Linux VM is BEHIND an Azure Load Balancer, timestamps MUST be set to 0
+net.ipv4.tcp_timestamps = 1
+
+# Reuse does require tcp_timestamps to be enabled. If tcp_timestamps are disabled because of load balancers, you should set reuse to 2.
+net.ipv4.tcp_tw_reuse = 1
+
+# Allowed local port range. This will increase the number of locally available ports (source ports)
+net.ipv4.ip_local_port_range = 1024 65535
+
+# Maximum number of packets taken from all interfaces in one polling cycle (NAPI poll). In one polling cycle interfaces which are # registered to polling are probed in a round-robin manner.
+net.core.netdev_budget = 1000
+
+# For high-performance environments, it's recommended to increase from the default 20KB to 65KB, in some extreme cases, for environments that support 100G+ networking, you can 
+# increase it to 1048576
+net.core.optmem_max = 65535
+
+# F-RTO is not recommended on wired networks. 
+net.ipv4.tcp_frto = 0
+
+# Increase the number of incoming connections / number of connections backlog
+net.core.somaxconn = 32768
+net.core.netdev_max_backlog = 32768
+net.core.dev_weight = 64
+```
+
+4. Optimize NIC ring buffers for TX/RX: Create an udev rule in ```/etc/udev/rules.d/99-azure-ring-buffer.rules``` to ensure they're applied to network interfaces:
+
+```plaintext
+# Setup Accelerated Interface ring buffers (Mellanox / Mana) 
+SUBSYSTEM=="net", DRIVERS=="hv_pci", ACTION=="add",  RUN+="/usr/sbin/ethtool -G $env{INTERFACE} rx 1024 tx 1024"
+
+# Setup Synthetic interface ring buffers (hv_netvsc)
+SUBSYSTEM=="net", DRIVERS=="hv_netvsc*", ACTION=="add",  RUN+="/usr/sbin/ethtool -G $env{INTERFACE} rx 1024 tx 1024"
+```
+
+For more information about optimizing network bandwidth, see [TCP/IP performance tuning for Azure VMs](/azure/virtual-network/virtual-network-tcpip-performance-tuning).
```

Comment:
Just needed some examples.  I love the DETAIL you had, but that needs to be in product documentation.  @bqtrinh, can you make sure this detail gets added to the right product documentation article?  It's too good to lose!

```suggestion
You should optimize the [Linux TCP/IP stack and buffers in the network interface](/azure/virtual-network/virtual-network-tcpip-performance-tuning) to achieve consistent performance. Follow Microsoft recommended settings. For example, you'll adjust items such as:

- Kernel parameters to optimize read and write memory buffers
- Bottleneck bandwidth and round-trip propagation time (BBR) congestion control
- Adjust TCP parameters to bring better consistency and throughput
- NIC ring buffers for TX/RX
```